### PR TITLE
Ol6

### PR DESCRIPTION
**Pull request by idovcohen**

### CODE REVIEW

File: src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts [hunk 1]
```diff
@@ -304,7 +304,9 @@ export class OpenLayersMap extends BaseImageryMap<OLMap> {
 	setMainLayerToBackgroundMap(layer: Layer) {
 		layer.set(ImageryLayerProperties.NAME, IMAGERY_MAIN_LAYER_NAME);
 		this.backgroundMapObject.getLayers().clear();
-		this.backgroundMapObject.addLayer(layer);
+		// TODO: Do we need to deep-copy the layer? Object.create does not do it...
+		const clonedLayer = Object.create(layer);
```
Comment (asafMasa) [inline src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts]: why do you need to clone or create it?

File: src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts [hunk 2]
```diff
@@ -333,6 +335,7 @@ export class OpenLayersMap extends BaseImageryMap<OLMap> {
 	public removeAllLayers() {
 		this.showGroups.forEach((show, group) => {
 			if (show && this._mapObject) {
+				// TODO: Should we remove the layer from the background map too?
```
Comment (asafMasa) [inline src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts]: no because we only add to it the main layer

File: src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts [hunk 3]
```diff
@@ -351,6 +354,8 @@ export class OpenLayersMap extends BaseImageryMap<OLMap> {
 		olShare.removeWorkers(layer);
 		this._mapLayers = this._mapLayers.filter((mapLayer) => mapLayer !== layer);
 		this._mapObject.removeLayer(layer);
+		// TODO: Should we remove the layer from the background map too?
```
Comment (asafMasa) [inline src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts]: no because we only add to it the main layer and we clear the background Map before new layer is added

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts [hunk 4]
```diff
@@ -342,6 +342,7 @@ export class MeasureDistanceVisualizer extends EntitiesVisualizer {
 
 	private setLabelsFeature() {
 		if (!this.measureData.meausres.length) {
+			// TODO: Do we want to get data from the server here? if so, use refresh()
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts]: we wanted to clear the source vector (when no items were received) not to refresh it

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/overlays/frame-visualizer.ts [hunk 5]
```diff
@@ -49,6 +49,7 @@ export class FrameVisualizer extends EntitiesVisualizer {
 			const features = this.source.getFeatures();
 			if (features && features[0]) {
 				delete (<any>features[0]).styleCache;
+				// TODO: Do we want to re-render the layer? if so, use changed().
 				this.source.refresh();
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/overlays/frame-visualizer.ts]: what are the differences between changed() and refresh()?

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts [hunk 6]
```diff
@@ -554,7 +554,7 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
 			this.updateFeature(featureId, { editMode: false });
 			this.source.removeFeature(this.edited.labelFeature);
 		}
-		this.source.refresh();
+		this.source.changed();
```
Comment (asafMasa) [inline src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: why change from refresh?

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts [hunk 7]
```diff
@@ -515,7 +515,7 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
 		if (entity) {
 			entity.originalEntity = merge({}, entity.originalEntity, props);
 			this.events.updateEntity.next(entity.originalEntity);
-			this.source.refresh();
+			this.source.changed();
```
Comment (asafMasa) [inline src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: why change from refresh?

File: src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts (hunk 1 above)
Comment (idovcohen) [reply src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts]: In OL6 we can't add the same layer to 2 different maps...

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/overlays/frame-visualizer.ts (hunk 5 above)
Comment (idovcohen) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/overlays/frame-visualizer.ts]: In OL6 'changed' is supposed to re-render the layer. They changed things a little bit...

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts (hunk 7 above)
Comment (asafMasa) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: double click on the map zooms in - we don't won't this behavior

Comment (asafMasa) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: image
maps seems black when changing between active maps

Comment (asafMasa) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: when zooming out on base map we see black tile - probably because of the deletion of:
loadTilesWhileInteracting: true, loadTilesWhileAnimating: true
we need to prevent this behavior - we should not see black tiles

Comment (idovcohen) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: Their documentation says that 'loadTilesWhileInteracting' and 'loadTilesWhileAnimating' are no longer used and it's supposed to be automatic, but I'll try...

Comment (idovcohen) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts]: I see this happening in the current version too...

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts [hunk 8]
```diff
@@ -268,12 +284,20 @@ export class MeasureDistanceVisualizer extends EntitiesVisualizer {
 	}
 
 	onDrawEndEvent(data) {
-		this.projectionService.projectCollectionAccurately([data.feature], this.iMap.mapObject)
+		data.feature.setId(UUID.UUID());
+		const measures = this.createMeasureLabelsFeatures(data.feature);
+		this.projectionService.projectCollectionAccurately([data.feature, ...measures], this.iMap.mapObject)
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:289]: pipe(take(1))

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts [hunk 9]
```diff
@@ -552,11 +515,16 @@ export class MeasureDistanceVisualizer extends EntitiesVisualizer {
...
+		// this.clearLabelInteractions();
+		this.removeTranslateMeasuresLabelInteraction();
 		this.removeInteraction(VisualizerInteractions.drawInteractionHandler);
 		this.removeInteraction(VisualizerInteractions.pointerMove);
 		this.removeInteraction(VisualizerInteractions.click);
 		super.onDispose();
 	}
+
+	private filterLineStringFeature(feature) {
+		return feature.getGeometry().getType() === 'LineString'
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:527]: ;

File: src/app/@ansyn/ansyn/package.json [hunk 10]
```diff
@@ -1,22 +1,21 @@
...
+		"@ansyn/map-facade": "3.0.0",
+		"@ansyn/menu": "3.0.0",
+		"@ansyn/ol": "3.0.0",
+		"@ansyn/imagery-cesium": "3.0.0",
+		"@ansyn/imagery-video": "3.0.0",
 		"@ansyn/event-drops": "^0.2.8-1.9",
-		"ol": "^5.3.0",
 		"@angular/cdk": "^7.3.0",
-		"@angular/material": "^7.3.0",
-		"@ngx-translate/core": "^11.0.1"
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/package.json:19]: Why did this package change from  peer dependency to dependency?

File: src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts [hunk 11]
```diff
@@ -70,7 +70,7 @@ export class MpTileImageSource extends ProjectableRaster {
 			tileUrlFunction: function (tileCoord, pixelRatio, proj) {
 				const z = tileCoord[0];
 				const x = tileCoord[1];
-				const y = -tileCoord[2] - 1;
```
Comment (ramtob) [inline src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts:73]: What does this change mean?

File: src/app/@ansyn/ol/mapSourceProviders/open-layers-ESRI-4326-source-provider.ts [hunk 12]
```diff
@@ -33,9 +31,9 @@ export class OpenLayerESRI4326SourceProvider extends OpenLayersMapSourceProvider
 				return config.baseUrl
 					.replace('{z}', (tileCoord[0] - 1).toString())
 					.replace('{x}', tileCoord[1].toString())
-					.replace('{y}', (-tileCoord[2] - 1).toString());
+					.replace('{y}', (tileCoord[2]).toString());
 			},
-			wrapX: true
```
Comment (ramtob) [inline src/app/@ansyn/ol/mapSourceProviders/open-layers-ESRI-4326-source-provider.ts:38]: Why this change?

File: src/app/@ansyn/ol/mapSourceProviders/open-layers.map-source-provider.ts [hunk 13]
```diff
@@ -52,7 +52,6 @@ export abstract class OpenLayersMapSourceProvider<CONF = any> extends BaseMapSou
 		});
 		const imageLayer = this.getImageLayer(source, extent);
 		this.removeExtraData(imageLayer);
-		tileLayer.set(IMAGE_PROCESS_ATTRIBUTE, imageLayer);
```
Comment (ramtob) [inline src/app/@ansyn/ol/mapSourceProviders/open-layers.map-source-provider.ts:55]: Was this line removed on purpose?

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts [hunk 14]
```diff
@@ -585,7 +585,7 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
 
 	labelTranslateMode(featureId: any) {
 		let oldFeature = null;
-		let event = null;
```
Comment (ramtob) [inline src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts:588]: Is this change needed?

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts [hunk 15]
```diff
@@ -9,52 +9,46 @@ import Circle from 'ol/style/Circle';
 import Point from 'ol/geom/Point';
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:9]: The file compare says that this file especially changed a lot. What is the reason for this?

**Review by ramtob — CHANGES_REQUESTED** [review]
Hi, I asked some questions about source changes

File: .gitignore [hunk 16]
```diff
@@ -32,6 +32,7 @@ node_modules
 !.vscode/tasks.json
 !.vscode/launch.json
 !.vscode/extensions.json
+.angulardoc.json
```
[the remark this answers is not in the document]
Comment (tltk90) [reply .gitignore:35]: is file that created from some extension of VS code,
ido has working with vs code so it add to our project,
we can remove this from `.gitignore`,
but it not necessary to us.

File: src/app/@ansyn/ansyn/ansyn/ansyn.component.ts [hunk 17]
```diff
@@ -58,7 +57,6 @@ export class AnsynComponent implements OnInit {
 	constructor(protected store$: Store<any>,
 				@Inject(COMPONENT_MODE) public componentMode: boolean,
 				@Inject(toolsConfig) public toolsConfigData: IToolsConfig,
-				protected ansynApp: ElementRef,
```
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/ansyn/ansyn.component.ts:61]: it was unused

File: src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts [hunk 18]
```diff
@@ -20,6 +20,7 @@ export interface IMeasureDataOptions {
 	isLayerShowed: boolean;
 	isToolActive: boolean;
 	isRemoveMeasureModeActive: boolean;
+	forceDisableTranslate?: boolean;
```
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts:23]: it come with merge from master,
i guess it disable the measure tools.

File: src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts (hunk 11 above)
Comment (tltk90) [reply src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts:73]: This is a change of Ido

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/layers/openlayers-arcgis-layers.pulgin.ts [hunk 19]
```diff
@@ -24,6 +24,7 @@ export class OpenlayersArcgisLayersPulgin extends OpenlayersBaseLayersPlugins {
 	createLayer(layer: ILayer): Observable<TileLayer> {
 		const extent: any = proj.transformExtent(layer.extent, EPSG_4326, this.iMap.mapObject.getView().getProjection());
 		const vector = new TileLayer({
+			preload: Infinity,
```
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/layers/openlayers-arcgis-layers.pulgin.ts:27]: it come with merge from master,
according to OL 

Preload. Load low-resolution tiles up to preload levels. 0 means no preloading.

File: src/app/@ansyn/ol/mapSourceProviders/open-layers-ESRI-4326-source-provider.ts (hunk 12 above)
Comment (tltk90) [reply src/app/@ansyn/ol/mapSourceProviders/open-layers-ESRI-4326-source-provider.ts:38]: is for not over draw this layer on his x-axis.

File: src/app/@ansyn/ol/mapSourceProviders/open-layers.map-source-provider.ts (hunk 13 above)
Comment (tltk90) [reply src/app/@ansyn/ol/mapSourceProviders/open-layers.map-source-provider.ts:55]: yes. i think the two line above also need to be removed.

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts (hunk 14 above)
Comment (tltk90) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts:588]: i guess, it come from master merge

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts (hunk 15 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:9]: there was problem with the OL v.6 some we change how the measure work.
maybe this led to many change in this file.

File: src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts:23]: But there is there word "translate" in this. Is this addition related to OL6? Perhaps there was some change in OL6 regarding Hebrew/RTL?

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/layers/openlayers-arcgis-layers.pulgin.ts (hunk 19 above)
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/layers/openlayers-arcgis-layers.pulgin.ts:27]: Yes, but why do we set it to infinity? The default is 0

File: src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts (hunk 11 above)
Comment (ramtob) [reply src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts:73]: So, should I ask Ido? :) This is a strange change. Do you have an idea why it was made?

File: src/app/@ansyn/ol/mapSourceProviders/open-layers.map-source-provider.ts (hunk 13 above)
Comment (ramtob) [reply src/app/@ansyn/ol/mapSourceProviders/open-layers.map-source-provider.ts:55]: Can u remove these lines then?

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts (hunk 14 above)
Comment (ramtob) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts:588]: What do you mean? There is here a change from `let event = null` to `let event = undefined`. What is the reason for this change?

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts (hunk 15 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:9]: Do you know what specifically was the problem with OL6 here, and what we changed in how the measure works?

Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:9]: @tltk90

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/layers/openlayers-arcgis-layers.pulgin.ts (hunk 19 above)
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/layers/openlayers-arcgis-layers.pulgin.ts:27]: in most of our application we use `preload: Infinity`

File: src/app/@ansyn/ansyn/app-effects/effects/cases.app.effects.ts [hunk 20]
```diff
@@ -129,15 +134,16 @@ export class CasesAppEffects {
...
 		);
 
-	@Effect()
-	onLoadDefaultCase$: Observable<IBaseImageryLayer> = this.actions$.pipe(
+	@Effect({dispatch: false})
+	onLoadDefaultCase$ = this.actions$.pipe(
 		ofType(CasesActionTypes.LOAD_DEFAULT_CASE),
 		withLatestFrom(this.store$.select(selectMapsIds)),
 		filter(([action, [mapId]]: [LoadDefaultCaseAction, string[]]) => !action.payload.context && Boolean(mapId)),
-		mergeMap(([action, [mapId]]: [LoadDefaultCaseAction, string[]]) => {
```
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/app-effects/effects/cases.app.effects.ts:137]: this effect was return an OL layer,
as return from `communicator.loadInitialMapSource(position)`
but in effect you should return an Action to be dispatch.
so no need to mergeMap in the end because i dont return an action,
and also dispatch: false was  require because i don't dispatch anything.

File: src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts (hunk 11 above)
Comment (tltk90) [reply src/app/@ansyn/ol/mapSourceProviders/marco/ol-utils/mp-tile-image-source.ts:73]: it was some breaking change from 5 to 6.
[see here](https://github.com/openlayers/openlayers/blob/main/changelog/upgrade-notes.md)

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts (hunk 15 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer.ts:9]: yes. we remove some interaction while ol iterate over the interaction array,
and it make some error for ol try call to function from undefined object.

so i change that all the measure act with one interaction instead that every measure create it's own interaction

File: src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts:23]: no.
if for that when you measure, and then click on annotation edit,
the measure will be disable.

File: src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts [hunk 21]
```diff
@@ -528,6 +527,8 @@ export class OpenLayersMap extends BaseImageryMap<OLMap> {
 
 	public addGeojsonLayer(data: GeoJsonObject): void {
 		let layer: VectorLayer = new VectorLayer({
+			updateWhileAnimating: true,
```
Comment (sulimani55) [inline src/app/@ansyn/ol/maps/open-layers-map/openlayers-map/openlayers-map.ts:530]: can you elaborate on there new lines?

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 22]
```diff
@@ -198,7 +189,7 @@ export class ToolsAppEffects {
 			mapIds.forEach((mapId) => {
 				const updateMeasureAction = new UpdateMeasureDataOptionsAction({
 					mapId: mapId,
-					options: { isToolActive: false }
+					options: { isToolActive: false, isRemoveMeasureModeActive: false}
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts:192]: can you elaborate on this new option?

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts [hunk 23]
```diff
@@ -26,6 +26,7 @@ const LOGS = {
 })
 @AutoSubscriptions()
 export class ExportMapsPopupComponent implements OnInit, OnDestroy {
+	animationFrame;
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:29]: isnt this redundent?

File: src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (sulimani55) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts:23]: what does translate mean in that context?

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/image-processing/image-processing.plugin.ts [hunk 24]
```diff
@@ -55,9 +55,8 @@ export class ImageProcessingPlugin extends BaseImageryPlugin {
 				return;
 			}
 			// else
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/image-processing/image-processing.plugin.ts:57]: is this else comment necessary?

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/north-calculations/north-calculations.plugin.ts [hunk 25]
```diff
@@ -1,4 +1,5 @@
 import { EMPTY, forkJoin, Observable, Observer, of, throwError } from 'rxjs';
+import { cloneDeep } from 'lodash';
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/plugins/north-calculations/north-calculations.plugin.ts:2]: this is not used

File: src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts (hunk 14 above)
Comment (tltk90) [reply src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts:588]: i remember. 
there is another place that use `event=undefined`,
and there is check of `event === undefined` in some place to decide if
the measure is active or not.
it was or change the above or the other to null.

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts (hunk 22 above)
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts:192]: there is an option to remove one measure
it's detect if the remove measure is active or nor.

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 26]
```diff
@@ -207,6 +198,13 @@ export class ToolsAppEffects {
 				clearActions = differenceWith(clearActions, action.payload.skipClearFor,
 					(act, actType) => act instanceof actType);
 			}
+			mapIds.forEach((mapId) => {
+				const updateMeasureAction = new UpdateMeasureDataOptionsAction({
+					mapId: mapId,
+					options: { forceDisableTranslate: undefined }
```
[the remark this answers is not in the document]
Comment (tltk90) [reply src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts:204]: when the user measure and then click on edit annotation.

File: src/app/@ansyn/ansyn/package.json (hunk 10 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/package.json:19]: because it more suitable to be dependency.
why should i force our client to run `npm i @ngx-translate/core`

**Review by ramtob — APPROVED** [review]
👍

**Review by sulimani55 — APPROVED** [review]
